holoviews/plotting/bokeh/util.py

```diff
@@ -407,6 +410,79 @@ def merge(tool, group):
 
     return Toolbar(tools=group_tools(tools, merge=merge, ignore=ignore) if merge_tools else tools)
 
+
+def sync_legends(plot_layout):
+    """This syncs the legends of all plots in a grid based on their name.
+
+    Only works for Bokeh 3 and above.
+
+    Parameters
+    ----------
+    plot_layout : bokeh.models.{GridPlot, Row, Column}
+        Gridplot to sync legends of.
+    """
+    if not bokeh3 or len(plot_layout.children) < 2:
+        return
+
+    # Collect all glyph with names
+    items = defaultdict(lambda: [])
+    click_policies = set()
+    for fig in plot_layout.children:
+        if isinstance(fig, tuple):  # GridPlot
+            fig = fig[0]
+        if not isinstance(fig, figure):
+            continue
+        for r in fig.renderers:
+            if r.name:
+                items[r.name].append(r)
+        if fig.legend:
+            click_policies.add(fig.legend.click_policy)
+
+    click_policies.discard("none")  # If legend is not visible, click_policy is "none"
+    if len(click_policies) > 1:
+        warn("Click policy of legends are not the same, no syncing will happen.")
+        return
+    elif not click_policies:
+        return
+
+    # Link all glyphs with the same name
+    mapping = {"mute": "muted", "hide": "visible"}
+    policy = mapping.get(next(iter(click_policies)))
+    code = f"dst.{policy} = src.{policy}"
+    for item in items.values():
+        for src, dst in permutations(item, 2):
+            src.js_on_change(
+                policy,
+                CustomJS(code=code, args=dict(src=src, dst=dst)),
+            )
+
+def one_legend(plot_layout, legend_no=0, legend_position="top_right"):
```

Comment:
What if we have legend_no as a list so users can have it displayed on multiple figures and maybe rename function as `limit_legends`? Also thoughts on renaming to `figure_index`?

Reply:
I'm not opposed to that idea. The function name would definitely need a rename. Not sure if `limit_legends` is the right one, though.